onnxruntime/core/providers/nuphar/scripts/symbolic_shape_infer.py

```diff
@@ -611,15 +611,25 @@ def _infer_OneHot(self, node):
         vi.CopyFrom(helper.make_tensor_value_info(node.output[0], self.known_vi_[node.input[2]].type.tensor_type.elem_type, new_shape))
 
     def _infer_Pad(self, node):
+        if get_opset(self.out_mp_) <= 10:
+            pads = get_attribute(node, 'pads')
+        else:
+            pads = self._try_get_value(node, 1)
+
         vi = self.known_vi_[node.output[0]]
-        if None in get_shape_from_type_proto(vi.type):
+        output_shape = get_shape_from_type_proto(vi.type)
+        if len(output_shape) == 0 or None in output_shape:
             sympy_shape = self._get_sympy_shape(node, 0)
             rank = len(sympy_shape)
             pads = get_attribute(node, 'pads')
             assert len(pads) == 2*rank
             new_shape = [d + pad_up + pad_down for d, pad_up, pad_down in zip(sympy_shape, pads[:rank], pads[rank:])]
             self._update_computed_dims(new_shape)
-            vi.CopyFrom(helper.make_tensor_value_info(node.output[0], vi.type.tensor_type.elem_type, get_shape_from_sympy_shape(new_shape)))
+            if vi.type.tensor_type.elem_type !=  onnx.TensorProto.UNDEFINED:
+                output_tp = vi.type.tensor_type.elem_type
+            else:
+                output_tp = self.known_vi_[node.input[0]].type.tensor_type.elem_type
```

Comment:
>self.known_vi_[node.input[0]] [](start = 28, length = 29)

why not always use input[0] elem_type in this line?

Reply:
Is it possible that input[0]'s elem_type is undefined but output[0]'s elem_type is valid?

Reply:
I don't think so

---
In reply to: [333771495](https://github.com/microsoft/onnxruntime/pull/2085#discussion_r333771495) [](ancestors = 333771495)

Reply:
Fixed.